### PR TITLE
Feat/slicing kedro viz

### PR DESCRIPTION
**Pull request by rashidakanchwala**
## Description

Experimentation on filtering pipelines by slicing them the way kedro does. 

kedro-viz-slice

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: src/selectors/filtered-pipeline.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+ */
+const getEdges = createSelector(
+  [getEdgeIDs, getEdgeSources, getEdgeTargets],
+  (edgeIDs, edgeSources, edgeTargets) =>
+    edgeIDs.map((id) => ({
+      id,
+      source: edgeSources[id],
+      target: edgeTargets[id],
+    }))
+);
```
Comment (tynandebold) [inline src/selectors/filtered-pipeline.js:22]: Is there any reason this logic can't live within the `getEdgesByNode` selector? It doesn't look like you're using it anywhere else.

File: src/selectors/filtered-pipeline.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+/**
+ * Selector to organize edges by their source and target nodes.
+ * @param {Array} edges - Array of edge objects.
+ * @returns {Object} An object containing edges mapped by source and target nodes.
+ */
+
+export const getEdgesByNode = createSelector([getEdges], (edges) => {
```
Comment (tynandebold) [inline src/selectors/filtered-pipeline.js:30]: Don't think you need to export this, do you?

File: src/selectors/filtered-pipeline.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      const linkedNodesBetween = [];
+      for (const nodeID in linkedNodesBeforeEnd) {
+        if (linkedNodesAfterStart[nodeID]) {
+          linkedNodesBetween.push(nodeID);
+        }
+      }
+
+      filteredNodeIDs = linkedNodesBetween;
+
+      return filteredNodeIDs;
```
Comment (tynandebold) [inline src/selectors/filtered-pipeline.js:104]: Do you need to reassign it here?

```suggestion
      return linkedNodesBetween;
```

**Review by tynandebold — COMMENTED** [review]
Just had a first look. I don't see anything major I would change. It's looking good! Can still get on a call to have you explain things in more detail though. I'm sure I'm missing something.

**Comment by rashidakanchwala** [comment]
I am closing this PR but leaving the branch for @Huongg to work from it.

**Pull request closed without merging**